### ISSUE TITLE
Better color theming support. More options and improve consistency.

### ISSUE THREAD

**Issue by matty-r** [apache/netbeans#6699]
### Description

I would love for Netbeans to support a much larger selection identifiers for selecting colors. Below is a screenshot of Netbeans and VSCode looking at the same method, note that VSCode has a much broader range of identifiers for choosing colors. Also note that the method calls are inconsistent. In the screenshot for example I would expect setUp(), createGUI(), addToPrivateRooms(), etc.. all to be the same color as in vscode.

![image](https://github.com/apache/netbeans/assets/9965448/6ec1d8f4-aff7-45de-8f22-8863a2cd9f0d)

### Use case/motivation

I find it very difficult to identify the differences between variables, method calls and other keyword identifiers.  It is also inconsistent which further exacerbates the issue. If there was more fine grained control on these colors it would vastly improve readability at a glance.

### Are you willing to submit a pull request?

Yes

**Comment by istinnstudio** [comment]
Sorry if I am wrong but all those features are there in fonts n colors - syntax - java. You have to setup your color/effect for each syntax element.

**Comment by matty-r** [comment]
> All those features are there in fonts n colors - syntax - java. You have to setup your color/effect for each syntax element. Only the preview (in the syntax tab) needs revision in my opinion as it does not "previewing" all those cases, making hard to believe that they are already there.

Unfortunately not. For example, i've selected "Identifier" here and set it to yellow, and "Keyword" is red - this covers about 90% of it in the shown example:

![image](https://github.com/apache/netbeans/assets/9965448/531c2623-1ba1-43b1-a90c-36b94deb8414)

**Comment by istinnstudio** [comment]
Everyone that needs full color on keywords should forget Identifier and change all the rest, the only issue I can see is the preview code itself and maybe a note to tell that identifier is a group of elements that can be easily "overwritten" in essence bypassed. At the end you can do what you see in vscode. For the close brackets: https://plugins.netbeans.apache.org/catalogue/?id=25

**Comment by matty-r** [comment]
There is no "all the rest"

**Comment by istinnstudio** [comment]
I sincerely do not know what you mean, all elements are in the list ready to be colorized. I do not see something that is missing. There is not automatic colorization of those elements, you have to do it one by one, for those you want to override inherited default coloring. There is not any missing functionality that you describe in screenshot 1.

**Comment by matty-r** [comment]
Well there are a few things wrong then, the selection doesn't highlight what color you're currently changing in the example, and the example is so narrow that it's extremely difficult to know what exactly you're changing.

1. Example code needs to be broadened to include a majority of the elements, or we should be able to paste in our own code
2. Selecting one of the elements needs to highlight/underline what you're currently changing to make it obvious, this currently doesn't happen for a majority of them
3. Clicking one of the elements in the example code should have a little popup to list all related colors for that element, so you can easily identify which color needs to be changed
4. It should be made obvious if a color is Inherited, what keyword/element it's Inherited from

**Comment by istinnstudio** [comment]
You are right, it is indeed confusing and looks/feels incomplete, 100% on all those suggestions. At least now you can do all the colorization one-by one it by just pressing OK, review in the real netbeans editor, re-open  the options window and continue. Frustrating but it works. So functionality is there but far from being easy,  especially the incomplete "preview" not showing the possible changes. I think "preview" should be in the title, in case anyone can create **at least a new preview code text** that can form/show all possible elements.

**Comment by matty-r** [comment]
I tried modifying the JavaExample and it didn't really help. I'll have a think on the title, but certainly the desired outcome of this request could be more focused based on our discussion.

**Comment by KaranAhlawat** [comment]
One thing I'd like to add is that LSP spec defines semantic tokens. Possibly, netbeans can implement this in their LSP client (?) and then nbcode itself can send these semantic tokens to clients, where they are handled appropriately. This would allow colorschemes to handle identifiers in a more consistent manner.

**Comment by lkishalmi** [comment]
NetBeans takes the syntax coloring from two main places. One is from the Lexer which is the most basic thing. It can recognize keywords, identifiers, separators, numbers, etc. The other source is from the parser where methods, classes fields, etc. can be recognized.

The preview dialog unfortunately does not support the parser based highlights.

**Comment by istinnstudio** [comment]
I guess someone who can do it, would change the preview to see the parser, and if it can be done, then it could let it free to accept a paste of any text.

**Comment by lkishalmi** [comment]
> I guess someone who can do it

Well, that could be a problem..

**Comment by matty-r** [comment]
> I guess someone who can do it, would change the preview to see the parser, and if it can be done, then preview could be free to accept any user text much like the main editor.

Well we may not need a preview version at all, we could just open a "tab" inside that preview window.

**Comment by Chris2011** [comment]
Why was this ticket closed? We should try to not say "not planned". Who did this label in general? There is no company behind that which has a backlog and says: This is nonsense, we are not doing it. We as a community needs to work on that. So it is a feature request and should stay open. I also miss this feature and it looks like this is a semantic thing for each language of NetBeams. So please do not use not planned to inflation. I'm for reopen it, let's stay it there and maybe use another label like "no responislbe person" or whatever. One day some people for hackathons are going through the issue section and see open tickets who wants to work on it. They will not see those feature request due to it was closed.

**Comment by Chris2011** [comment]
I see, We already have one "contribution welcome" and this should be used instead. IMHO.
